aws/table_aws_sfn_state_machine_execution_history.go

```diff
@@ -250,47 +256,64 @@ func listStepFunctionsStateMachineExecutionHistories(ctx context.Context, d *plu
 		return nil, nil
 	}
 
-	stateMachineArn := h.Item.(types.StateMachineListItem).StateMachineArn
-	var executions []types.ExecutionListItem
-	maxLimit := int32(1000)
-	// If the requested number of items is less than the paging max limit
-	// set the limit to that instead
-	limit := d.QueryContext.Limit
-	if d.QueryContext.Limit != nil {
-		if *limit < int64(maxLimit) {
-			maxLimit = int32(*limit)
+	var executions []string
+
+	plugin.Logger(ctx).Trace("aws_sfn_state_machine_execution_history.listStepFunctionsStateMachineExecutionHistories", fmt.Sprintf("d.Quals=%#v", d.Quals))
+	executionArnQuals := getQualsValueByColumn(d.Quals, "execution_arn", "string") // FIXME: this does not filter on operators at all, so issues other operators than `=` or `in` would occur, e.g. with `like`
+	plugin.Logger(ctx).Debug("aws_sfn_state_machine_execution_history.listStepFunctionsStateMachineExecutionHistories", "execution_arn quals", executionArnQuals)
+
+	// Minimize the API call with the given execution ARN
+	if executionArnQuals != nil {
+		if executionArnQualsStr, ok := executionArnQuals.(string); ok && executionArnQualsStr != "" {
+			executions = []string{executionArnQualsStr}
+
+		} else if executionArnQualsList, ok := executionArnQuals.([]string); ok && len(executionArnQualsList) > 0 {
+			executions = executionArnQualsList
+		}
+	} else {
+		stateMachineArn := h.Item.(types.StateMachineListItem).StateMachineArn
+		maxLimit := int32(1000)
+		// If the requested number of items is less than the paging max limit
+		// set the limit to that instead
+		limit := d.QueryContext.Limit
+		if d.QueryContext.Limit != nil {
+			if *limit < int64(maxLimit) {
+				maxLimit = int32(*limit)
+			}
+		}
+		input := &sfn.ListExecutionsInput{
+			MaxResults:      int32(maxLimit), //FIXME: the limit should apply to execution history items, not executions
```

Comment:
I believe this limit filter should be moved to the `GetExecutionHistory` call https://github.com/turbot/steampipe-plugin-aws/pull/1684/files#diff-ce56aaf58176fa3daec5808d3d4174d16d7d76d3c9ff224ed859e9baa3c3f998R368